Easydict/NewApp/View/SettingView/Tabs/DisabledAppTab.swift

```diff
@@ -0,0 +1,224 @@
+        }
+    }
+
+    func appendNewDisabledApp(for url: URL) {
+        guard let selectAppModel = disabledAppModel(from: url) else { return }
+        guard !appModelList.contains(selectAppModel) else { return }
+        appModelList.append(selectAppModel)
+        saveDisabledApps()
+    }
+
+    func disabledAppModel(from url: URL) -> EZAppModel? {
+        let appModel = EZAppModel()
+        guard let bundle = Bundle(url: url) else { return nil }
+        appModel.appBundleID = bundle.bundleIdentifier ?? ""
+        appModel.triggerType = []
+        return appModel
+    }
+}
+
+@available(macOS 13.0, *)
+struct DisabledAppTab: View {
+    @StateObject var disabledAppViewModel = DisabledAppViewModel()
+
+    var listToolbar: some View {
+        ListToolbar()
+            .fileImporter(
+                isPresented: $disabledAppViewModel.isImporting,
+                allowedContentTypes: [.application],
+                allowsMultipleSelection: true
+            ) { result in
+                switch result {
+                case let .success(urls):
+                    disabledAppViewModel.newAppURLsSelected(from: urls)
+                case let .failure(error):
+                    print("fileImporter error: \(error)")
+                    disabledAppViewModel.isShowImportErrorAlert.toggle()
+                }
+            }
+            .alert(isPresented: $disabledAppViewModel.isShowImportErrorAlert) {
+                Alert(title: Text(""), message: Text("setting.disabled.import_app_error.message"), dismissButton: .default(Text("ok")))
+            }
+    }
+
+    var appListView: some View {
+        List(selection: $disabledAppViewModel.selectedAppModels) {
+            ForEach(disabledAppViewModel.appModelList, id: \.self) { app in
+                BlockAppItemView(with: app)
+                    .tag(app)
+            }
+            .listRowSeparator(.hidden)
+        }
+        .listStyle(.plain)
+        .scrollIndicators(.never)
+    }
+
+    var appListViewWithToolbar: some View {
+        VStack(spacing: 0) {
+            appListView
+
+            listToolbar
+        }
+        .clipShape(RoundedRectangle(cornerRadius: 8))
+        .padding(.bottom)
+        .padding(.horizontal, 35)
+        .onTapGesture {
+            disabledAppViewModel.selectedAppModels = []
+        }
+    }
+
+    var body: some View {
+        VStack {
+            Text("disabled_title")
+                .padding()
+
+            appListViewWithToolbar
+        }
+        .frame(maxWidth: 500)
+        .environmentObject(disabledAppViewModel)
+    }
+}
+
+@available(macOS 13.0, *)
+private struct ListToolbar: View {
+    @EnvironmentObject private var disabledAppViewModel: DisabledAppViewModel
+
+    var body: some View {
+        VStack(spacing: 0) {
+            Divider()
+            HStack(spacing: 0) {
+                ListButton(imageName: "plus") {
+                    disabledAppViewModel.isImporting.toggle()
```

Comment:
I had this drag causing lag problem before and solved it in a very dumb way, maybe you can optimize this ugly code.

```objc
EZConfiguration.shared.disabledAutoSelect = YES;
...
add a disabled App 
...
EZConfiguration.shared.disabledAutoSelect = NO;
```

https://github.com/tisfeng/Easydict/blob/2df80b922865103b24d19a22b541b89c3218bf3e/Easydict/Feature/PerferenceWindow/DisableAutoSelectTextViewController/EZDisableAutoSelectTextViewController.m#L257

Reply:
Yes, I think this is the reason of first issue.

Reply:
Ok, I will open a new issue to track it.